pkg/cloudprovider/metrics/cloudprovider.go

```diff
@@ -70,31 +93,93 @@ func Decorate(cloudProvider cloudprovider.CloudProvider) cloudprovider.CloudProv
 }
 
 func (d *decorator) Create(ctx context.Context, machine *v1alpha5.Machine) (*v1alpha5.Machine, error) {
-	defer metrics.Measure(methodDurationHistogramVec.WithLabelValues(injection.GetControllerName(ctx), "Create", d.Name()))()
-	return d.CloudProvider.Create(ctx, machine)
+	method := "Create"
```

Comment:
What are your thoughts on using reflection to infer the method string for all of these functions? This will ensure the metrics map to the function names correctly even if the function names change.

Reply:
The other problem is that that reflection would be done at runtime which is notoriously slow. I really don't want to increase our latency based on a decorator that is going to increase the latency across _all_ method calls to cloudprovider